firefly_client/plot.py

```diff
@@ -0,0 +1,342 @@
+    nbins: `int`
+        number of bins to use for histogram. Default 30
+    title: `str`
+        plot title. Defaults to empty string
+    xlabel: `str`
+        x-axis label. If None, defaults to xname
+    ylabel: `str`
+        y-axis label. If None, defaults to yname
+    cell_id: `str`
+        ID of Slate cell from add_cell, defaults to table_cellid
+    """
+    layout = dict(xaxis=dict(title=xlabel if xlabel else data_col),
+                  yaxis=dict(title=ylabel if ylabel else 'Number'))
+    if title is not None:
+        layout['title'] = title
+    if len(tbl_id) == 0:
+        logger.debug('Using last_tblid: {}'.format(last_tblid))
+        tbl_id = last_tblid
+    hist_data = dict(
+        type='fireflyHistogram',
+        name=data_col,
+        marker={'color': 'rgba(153, 51, 153, 0.8)'},
+        firefly=dict(
+            tbl_id=tbl_id,
+            options=dict(
+                algorithm='fixedSizeBins',
+                fixedBinSizeSelection='numBins',
+                numBins=nbins,
+                columnOrExpr=data_col
+            )
+        )
+    )
+    hist_data.update(kwargs)
+    fc.show_chart(group_id=cell_id, layout=layout, data=[hist_data] )
+    return
+
+def upload_table(table, title=None, show=True, write_func="auto", tbl_index=1, page_size=200):
+    """upload a table object to Firefly
+
+    Parameters:
+    -----------
+    table: `str` or table-like object
+        path to table file, or table object to upload
+    title: `str`
+        title of the table, also used as the table ID, if specified. Otherwise auto-generated.
+    show: `bool`
+        display the table in a table viewer in the browser. default True.
+    write_func: `str` or function
+        function for serializing the table. If 'auto', try to auto-discover the write method
+        from the type of the table object and known table types.
+    tbl_index: `int`
+        If the table object contains multiple tables, use the first one.
+    page_size: `int`
+        Number of rows in each page of the table viewer. Default 200.
+
+    Returns:
+    --------
+    tbl_id: `str`
+        Table ID on the Firefly server
+    """
+    if isinstance(table, str):
+        tval = fc.upload_file(table)
+    else:
+        if write_func == 'auto':
+            if 'lsst.afw.table' in str(type(table)):
+                write_func = table.writeFits
+            elif 'astropy.table.table.Table' in str(type(table)):
+                def write_astropy(fname):
+                    atable = table.copy()
+                    for c in atable.colnames:
+                        if len(c) > 68:
+                            atable.rename_column(c, c[:68])
+                    #atable.remove_columns([c for c in table.colnames if len(c) > 68])
+                    import warnings
+                    with warnings.catch_warnings():
+                        warnings.simplefilter('ignore', UserWarning)
```

Comment:
Do we want to filter out UnitsWarning? A table column units should  not be restricted by FITS standards. Currently I get warnings like `UnitsWarning: 'e-/SUR ADU' did not parse as fits unit: At col 0, Unit 'e' not supported by the FITS standard.` when uploading an astropy version of our WISE table.

Reply:
I see that `UnitsWarning` is an `AstropyWarning` and not an `AstropyUnitsWarning`. My filter on `UserWarning` doesn't work on it. I suppose if I have an Astropy table in my Python session, by definition the astropy package is available and I can import `AstropyWarning`.